src/main/java/marquez/common/models/DatasetUrn.java

```diff
@@ -14,45 +14,22 @@
 
 package marquez.common.models;
 
-import java.util.StringJoiner;
 import java.util.regex.Pattern;
 import lombok.EqualsAndHashCode;
-import lombok.Getter;
 import lombok.NonNull;
-import lombok.ToString;
 
-@EqualsAndHashCode
-@ToString
-public final class DatasetUrn {
-  private static final int URN_MIN_SIZE = 1;
-  private static final int URN_MAX_SIZE = 64;
-  private static final String URN_DELIM = ":";
-  private static final String URN_PREFIX = "urn";
-  private static final String URN_REGEX =
-      String.format(
-          "^%s(%s[a-zA-Z0-9._]{%d,%d}){2}$", URN_PREFIX, URN_DELIM, URN_MIN_SIZE, URN_MAX_SIZE);
-  private static final Pattern URN_PATTERN = Pattern.compile(URN_REGEX);
+@EqualsAndHashCode(callSuper = false)
+public final class DatasetUrn extends Urn {
+  private static final int NUM_COMPONENTS = 2;
+  private static final String URN_TYPE = "dataset";
```

Comment:
As we define URNs, it'd be great to have a list known of types. Thoughts on defining the enum `UrnType`? So, initially, we'll have `UrnType.DATASET`, `UrnType.DATASOURCE`

Reply:
Hm, since we are creating an actual `*Urn` type for each urn type, does the enum provide much there? Adding a new Urn subtype would then require a change in the base `Urn`.